### PR TITLE
Add `isFailure` option

### PR DESCRIPTION
**Pull request by SimenB**
This copies the idea (and description) from Levee.

https://github.com/krakenjs/levee#isfailure

### CODE REVIEW

File: lib/Circuit.js [hunk 1]
```diff
@@ -64,28 +68,28 @@ class Circuit extends EventEmitter {
...
-          // there are no listeners or stats collection for
-          // the fallback function. The function is fire-and-forget
-          // as far as `Brakes` is concerned
-          if (this._fallback) {
-            return this._fallback.apply(this, arguments);
-          }
-
-          return Promise.reject(err);
-        });
+      .apply(this, arguments)
```
Comment (SimenB) [inline lib/Circuit.js]: Indent was odd wrong here

File: lib/Circuit.js [hunk 2]
```diff
@@ -64,28 +68,28 @@ class Circuit extends EventEmitter {
...
+      .tap(() => this._brakes.emit('success', Date.now() - startTime))
```
Comment (SimenB) [inline lib/Circuit.js]: http://bluebirdjs.com/docs/api/tap.html

**Comment by awolden** [comment]
@SimenB This looks like a solid addition. I see now issue release this as a minor version upgrade. I will merge and publish new version @ 2.4.0.

**Comment by SimenB** [comment]
@awolden Hmm, it probably shouldn't reject at all

**Comment by awolden** [comment]
It would seem strange to have it not reject. If the `isFailure()` function determines the error is not critical, it resolves the request? What would it return in the resolution?

**Comment by SimenB** [comment]
It would resolve with `null` I suppose
By providing `isFailure`, and having it return `false` you explicitly say it's not a failure, thus it shouldn't reject.

If a fallback is defined resolve with that, otherwise just resolve?

**Comment by SimenB** [comment]
@awolden Thoughts in ^? I can update tomorrow if you agree

**Comment by awolden** [comment]
I think it makes since as written. Levee still triggers the callback with the `err` as the first argument, essentially rejecting it. I think everything makes sense as written, and doesn't require any changes.

**Comment by SimenB** [comment]
Hmmm, true.
Merge away, then!

**Pull request merged**